client/src/components/UploadFileButton.tsx

```diff
@@ -0,0 +1,33 @@
+import { styled } from "@mui/material/styles";
+import Button from "@mui/material/Button";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+
+const VisuallyHiddenInput = styled("input")({
+  clip: "rect(0 0 0 0)",
```

Comment:
tych styli jest tu całkiem dużo i trochę dziwnych... 'clip` i `clipPath`? to są style z SVG.
whitespace, mimo że nie masz tekstu.
position:absolute w ogóle jest dziwne, bo jego parent nie ma position: relative.

Ogólnie to wydaje mi się, że wystarczy albo `visibility: hidden`, albo po prostu propka `hidden` na inpucie, tzn. 
```
<input hidden type="file"/>